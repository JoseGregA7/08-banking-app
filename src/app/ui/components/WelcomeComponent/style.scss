
$primary-gradient: linear-gradient(135deg, rgba(16, 65, 226, 0.589), rgb(201, 42, 14));
$background-color: #f4f4f4;
$primary-color: #007bff;
$text-color: #333;
$sub-text-color: #666;
$border-radius: 5px;

.buttons-container {
  display: flex;
  flex-direction: column;
  justify-content: end;
  align-items: center;
  gap: 10px;
  margin-bottom: 20px;
  width: 100%;
  padding: 4px;
}

.button {
  padding: 8px 4px;
  font-size: 0.8rem;
  background: $primary-gradient;
  color: #fff;
  border: none;
  border-radius: $border-radius;
  cursor: pointer;
  transition: background-color 0.3s;
  width: 150px;
  &:hover {
    opacity: 0.8;
  }
}

.home__forgot-password {
  color: $primary-color;
  text-decoration: underline;
  font-size: 0.8rem;
  margin-top: 10px;
}

.home__terms-conditions {
  color: black;
  font-size: 0.8rem;
  opacity: 0.8;
  margin-bottom: 20px;
}
