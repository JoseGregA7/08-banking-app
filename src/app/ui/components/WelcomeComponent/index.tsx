import './style.scss';

const WelcomeComponent = ({ handleLoginClick, handleSignUpClick }: { handleLoginClick: () => void, handleSignUpClick: () => void }) => {
    return (
        <div className='buttons-container'>
            <div>
                Ya tienes cuenta?
            </div>
            <button className='button' onClick={handleLoginClick}>Ingresar</button>
            <div>
                Unete ahora
            </div>
            <button className='button' onClick={handleSignUpClick}>Regístrate</button>

            <div>
                <a href="#" className='home__forgot-password'>
                    Olvide mi contraseña
                </a>
            </div>
        </div>
    )
}

export default WelcomeComponent;