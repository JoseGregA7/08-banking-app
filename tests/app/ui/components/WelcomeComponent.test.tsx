import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import WelcomeComponent from '../../../../src/app/ui/components/WelcomeComponent'; 
import { describe, expect, test, vitest } from 'vitest';

describe('WelcomeComponent', () => {
  test('should render the component with the expected buttons and links', () => {
    const handleLoginClick = vitest.fn();
    const handleSignUpClick = vitest.fn();
    
    const { getByText, getByRole } = render(
      <WelcomeComponent 
        handleLoginClick={handleLoginClick} 
        handleSignUpClick={handleSignUpClick} 
      />
    );
    
    expect(getByText('Ya tienes cuenta?')).toBeInTheDocument();
    expect(getByText('Unete ahora')).toBeInTheDocument();
    expect(getByText('Olvide mi contraseña')).toBeInTheDocument();
    expect(getByRole('button', { name: 'Ingresar' })).toBeInTheDocument();
    expect(getByRole('button', { name: 'Regístrate' })).toBeInTheDocument();
  });

  test('should call handleLoginClick when the "Ingresar" button is clicked', () => {
    const handleLoginClick = vitest.fn();
    const handleSignUpClick = vitest.fn();
    
    const { getByRole } = render(
      <WelcomeComponent 
        handleLoginClick={handleLoginClick} 
        handleSignUpClick={handleSignUpClick} 
      />
    );
    
    fireEvent.click(getByRole('button', { name: 'Ingresar' }));
    
    expect(handleLoginClick).toHaveBeenCalledTimes(1);
  });

  test('should call handleSignUpClick when the "Regístrate" button is clicked', () => {
    const handleLoginClick = vitest.fn();
    const handleSignUpClick = vitest.fn();
    
    const { getByRole } = render(
      <WelcomeComponent 
        handleLoginClick={handleLoginClick} 
        handleSignUpClick={handleSignUpClick} 
      />
    );
    
    fireEvent.click(getByRole('button', { name: 'Regístrate' }));
    
    expect(handleSignUpClick).toHaveBeenCalledTimes(1);
  });

  test('should have correct classes applied to elements', () => {
    const handleLoginClick = vitest.fn();
    const handleSignUpClick = vitest.fn();
    
    const { getByRole, getByText } = render(
      <WelcomeComponent 
        handleLoginClick={handleLoginClick} 
        handleSignUpClick={handleSignUpClick} 
      />
    );

    expect(getByRole('button', { name: 'Ingresar' })).toHaveClass('button');
    expect(getByRole('button', { name: 'Regístrate' })).toHaveClass('button');
    expect(getByText('Olvide mi contraseña')).toHaveClass('home__forgot-password');
  });

  test('should match snapshot', () => {
    const handleLoginClick = vitest.fn();
    const handleSignUpClick = vitest.fn();
    
    const { asFragment } = render(
      <WelcomeComponent 
        handleLoginClick={handleLoginClick} 
        handleSignUpClick={handleSignUpClick} 
      />
    );
    
    expect(asFragment()).toMatchSnapshot();
  });
});
