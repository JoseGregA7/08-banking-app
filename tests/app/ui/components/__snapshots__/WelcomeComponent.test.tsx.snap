// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WelcomeComponent > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="buttons-container"
  >
    <div>
      Ya tienes cuenta?
    </div>
    <button
      class="button"
    >
      Ingresar
    </button>
    <div>
      Unete ahora
    </div>
    <button
      class="button"
    >
      Regístrate
    </button>
    <div>
      <a
        class="home__forgot-password"
        href="#"
      >
        Olvide mi contraseña
      </a>
    </div>
  </div>
</DocumentFragment>
`;
